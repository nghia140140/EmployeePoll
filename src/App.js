import logo from "./logo.svg";
import "./App.css";
import { Route, Routes } from "react-router-dom";
import LoginPage from "./components/login";
import Home from "./components/home";
import LeaderBoard from "./components/leaderBoard";
import New from "./components/new";
import Menu from "./components/menu";
import { connect, useDispatch } from "react-redux";
import { useEffect } from "react";
import { getQuestions, getUsers } from "./redux/middleware/thunk";
import QuestionDetail from "./components/detailQuestion";
import Protected from "./components/protected";

function App({ authReducer }) {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getUsers());
    dispatch(getQuestions());
  }, []);

  return (
    <div>
      {!!authReducer && <Menu />}
      <Routes>
        <Route exact path="/" element={<LoginPage />} />
        <Route
          path="/home"
          element={
            <Protected>
              <Home />
            </Protected>
          }
        />
        <Route
          path="/question/:id"
          element={
            <Protected>
              <QuestionDetail />
            </Protected>
          }
        />

        <Route
          path="/leaderboard"
          element={
            <Protected>
              <LeaderBoard />
            </Protected>
          }
        />
        <Route
          path="/add"
          element={
            <Protected>
              <New />
            </Protected>
          }
        />
      </Routes>
    </div>
  );
}

const mapStateToProps = ({ authReducer }) => ({
  authReducer: authReducer,
});

export default connect(mapStateToProps)(App);
