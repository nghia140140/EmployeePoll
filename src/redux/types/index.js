export * from "./userTypes";
export * from "./auth";
export * from "./questionTypes";
