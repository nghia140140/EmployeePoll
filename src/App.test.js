import { render, screen } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "./redux/store";

describe("App", () => {
  test("should render login", async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );

    const loginTitle = screen.getByText("Login");
    expect(loginTitle).toBeInTheDocument();
  });
});
