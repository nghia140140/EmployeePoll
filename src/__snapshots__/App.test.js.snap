// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <h1>
            Login
          </h1>
          <img
            class="login-img"
            src="userLogin.png"
          />
          <form>
            <div>
              <h5>
                User
              </h5>
              <div
                class="inputUsername"
              >
                <input
                  id="username"
                  name="username"
                  placeholder="User"
                  type="text"
                  value="mtsamis"
                />
              </div>
            </div>
            <div>
              <h5>
                PassWord
              </h5>
              <div
                class="inputPassword"
              >
                <input
                  id="password"
                  name="password"
                  placeholder="Password"
                  type="text"
                  value="xyz123"
                />
              </div>
            </div>
            <button
              class="btnLogin"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <h1>
          Login
        </h1>
        <img
          class="login-img"
          src="userLogin.png"
        />
        <form>
          <div>
            <h5>
              User
            </h5>
            <div
              class="inputUsername"
            >
              <input
                id="username"
                name="username"
                placeholder="User"
                type="text"
                value="mtsamis"
              />
            </div>
          </div>
          <div>
            <h5>
              PassWord
            </h5>
            <div
              class="inputPassword"
            >
              <input
                id="password"
                name="password"
                placeholder="Password"
                type="text"
                value="xyz123"
              />
            </div>
          </div>
          <button
            class="btnLogin"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
